answer.js

```diff
@@ -1,17 +1,177 @@
 // Answer here for tasks.md
 
-// Q1 Answer here...
+// 1. create a user schema with following fields and type
+//   1.name -> string
+//   2.email -> string
+//   3.age -> number
 
-// Q2 Answer here...
+var userSchema = new Schema({
+    name: String,
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+        match: /@/
+    },
+    age: Number,
+});
 
-// Q3 Answer here...
 
-// Q4 Answer here...
+// Q2 update user schema to make email lowercase and default age to be 0.
 
-// Q5 Answer here...
+var userSchema = new Schema({
+    name: String,
+    email: {
+        type: String,
+        required: true,
+        lowercase: true,
+        match: /@/
+    },
+    age: {
+        type: Number,
+        default: 0
+    }
+})
 
-// Q6 Answer here...
+//3. Update user schema 
+// - contains password field with minimum 5 characters
+// - add createdAt field of type date and default it to current date.
 
-// Q7 Answer here...
+var userSchema = new Schema({
+    password: { 
+        type: String, 
+        required: true,
+        minlength: 5, 
+    },
+    createdAt: {
+        type: Date, 
+        default: new Date(),
+    }
+})
 
-// Q8 Answer here...
\ No newline at end of file
+
+// Q4 Add address Schema with fields
+//   1. village -> String
+//   2. city -> String
+//   3. state -> String
+//   4. pin -> number
+//   5. user -> ObjectId of User from user schema
+
+var addressSchema = new Schema ({
+    village: String,
+    city: String,
+    state: String,
+    pin: Number,
+    user: {
```

Comment:
user type should be `Schema.Types.ObjectId` and ref is always model name in string like
```js
var Schema = mogoose.Schema;
user: {
  type: Schema.Types.ObjectId,
  ref: 'User'
}
```